### PR TITLE
Switch to pure-julia types

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

**Comment by rdeits** [comment]
done!

**Comment by rdeits** [comment]
Ok, I think is ready, pending a new LCMCore tag.

**Comment by tkoolen** [comment]
Awesome, thanks so much!

File: src/sensor_status_t.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# The sensor_status_t has a field named `type`, which is a reserved
+# keyword in Julia v0.7 and below. To get around that, we can just
+# construct the expression directly instead of relying on the parser.
```
Comment (tkoolen) [inline src/sensor_status_t.jl:3]: That's unfortunate. Nice job working around this limitation.

Comment (rdeits) [reply src/sensor_status_t.jl:3]: Heh, yeah. It's actually not so bad (I can't imagine trying to do this in Python if we had an lcmtype with `class` as a field...).

Comment (rdeits) [reply src/sensor_status_t.jl:3]: Plus, in v1.0 we can get rid of this entirely!

**Pull request merged**